tools/plugin/modules/alsa.c

```diff
@@ -0,0 +1,790 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+
+/* file component for reading/writing pcm samples to/from a file */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <rtos/sof.h>
+#include <sof/list.h>
+#include <sof/audio/stream.h>
+#include <sof/audio/ipc-config.h>
+#include <sof/ipc/driver.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <ipc/dai.h>
+
+#include <alsa/asoundlib.h>
+
+#include "pipe.h"
+
+/* 66def9f0-39f2-11ed-89f7-af98a6440cc4 */
+DECLARE_SOF_RT_UUID("arecord", arecord_uuid, 0x66def9f0, 0x39f2, 0x11ed,
```

Comment:
These names are bit misleading as "arecord" is just one ALSA app and this plugin will work with any ALSA application that will record audio.  I guess this doesn't hurt either, but may mislead a bit...